sotodlib/io/load_smurf.py

```diff
@@ -656,11 +656,21 @@ def add_file(self, path, session):
                              )
         
             data = frame.get('data')
+            sostream_version = frame.get('sostream_version', 0)
             if data is not None:
-                db_frame.n_samples = data.n_samples
-                db_frame.n_channels = len(data)
-                db_frame.start = dt.datetime.fromtimestamp(data.start.time / spt3g_core.G3Units.s)
-                db_frame.stop = dt.datetime.fromtimestamp(data.stop.time / spt3g_core.G3Units.s)
+                if sostream_version >= 2:  # Using SuperTimestreams
+                    db_frame.n_channels, db_frame.n_samples = data.data.shape
```

Comment:
Accessing data.data causes the array to be decompressed, which is otherwise unnecessary.  Better to just use len(data.names), len(data.times).